backend/src/middleware/errorHandler.ts

```
@@ -0,0 +1,280 @@
+    };
+  }
+
+  res.status(error.statusCode).json(response);
+};
+
+// Handle Prisma errors
+const handlePrismaError = (error: Prisma.PrismaClientKnownRequestError): ApiError => {
+  switch (error.code) {
+    case 'P2002':
+      // Unique constraint violation
+      const field = error.meta?.['target'] as string[] | undefined;
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)